examples/ui-demo/src/hooks/useMintDefault.tsx

```diff
@@ -0,0 +1,149 @@
+import { useQuery } from "@tanstack/react-query";
+import { useCallback, useMemo, useState } from "react";
+import { useToast } from "./useToast";
+import {
+  useSmartAccountClient,
+  useSendUserOperation,
+} from "@account-kit/react";
+import { Address, encodeFunctionData } from "viem";
+import { AccountKitNftMinterABI } from "@/utils/config";
+import { MintStatus } from "@/components/small-cards/MintCard";
+
+const initialValuePropState = {
+  signing: "initial",
+  gas: "initial",
+  batch: "initial",
+} satisfies MintStatus;
+
+export interface UseMintReturn {
+  isLoading: boolean;
+  status: MintStatus;
+  mintStarted: boolean;
+  handleCollectNFT: () => unknown;
+  uri?: string;
+  transactionUrl?: string;
+}
+
+// TODO(jh): Once the client supports switching b/w default
+// & 7702 modes, this hook should support both modes.
+export const useMintDefault = (props: {
+  contractAddress: Address;
```

Comment:
once a client can switch MAv2 modes, this will be renamed to `useMint` and take a `mode` prop. then the `useMint7702` hook will be removed.